pkg/image/apis/image/validation/whitelist/whitelister.go

```diff
@@ -57,7 +64,7 @@ type allowedHostPortGlobs struct {
 type registryWhitelister struct {
 	whitelist             []allowedHostPortGlobs
 	pullSpecs             sets.String
-	registryHostRetriever imageapi.RegistryHostnameRetriever
+	registryHostRetriever RegistryHostnameRetriever
```

Comment:
> circular dependency issue?

general dependency issue.  Golang encourages the creation of local interfaces to avoid package entanglement and take advantage of duck typing without hard deps.  Taking a dependency on this package ends up bring in the entire client which affects the size of binaries only dealing with config which clayton wants small.

Also, this is more idiomatic.

Reply:
i guess that makes sense, though i'm not a fan of scattered copies of the same (or slightly differing but related versions) interfaces.  Seems like a good way to get yourself confused and out of sync.

Anyway ok